### PR TITLE
Add Settings Button and configuration

### PR DESCRIPTION
**Pull request by RazeeHJ**

### CODE REVIEW

File: solutions/devsprint-riccardo-1/OnboardingChallenge.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -457,7 +450,7 @@
 				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
-				DEVELOPMENT_TEAM = B8F644M47X;
+				DEVELOPMENT_TEAM = WA9JUKEPEP;
```
Comment (cipolleschi) [inline solutions/devsprint-riccardo-1/OnboardingChallenge.xcodeproj/project.pbxproj:453]: could you revert the change done to the development team? 🙏 Otherwise everyone will get the wrong team in the project.

File: solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListViewController.swift [hunk 2]
```diff
@@ -3,74 +3,92 @@
...
+  private func fetchList() {
+    self.service.fetchList { items in
+      let configuration = ListViewConfiguration(listItems: items)
+      self.listView.updateView(with: configuration)
     }
+  }
+
+  @objc private func showSettings() {
+    navigationController?.present(
+      SettingsViewController(interactor: SettingsInteractor(bundleService: Bundle.main)),
```
Comment (cipolleschi) [inline solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListViewController.swift:82]: here we are coupling together the `ListViewController` with the navigation to the `SettingsViewController`. In a modularized architecture, it means that the `ListViewController` module must know about the `SettingsViewController`. That can create issues because if the `SettingsViewController` changes its interface, for example, the `ListViewController` must change as well.

This is a violation of the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle): we don't want the ListViewController to change due to a change in the Settings.

Can you think about a way to decouple the two view controller? 



_hint_: it is similar to what we did with the `BundleService`, asking it from the outside...

**Review by cipolleschi — CHANGES_REQUESTED** [review]
Great job, almost perfect. There is a change in the project file that we need to fix and we can improve the app architecture a little bit!

File: solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListViewController.swift (hunk 2 above)
Comment (RazeeHJ) [reply solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListViewController.swift:82]: Would a router that handles routing from ListViewController to SettingsViewController be solving this violation? 

Or would you recommend a router that starts the transfer and then another router that sort of accepts it? So like two routers, one belongs to List and the other one to Settings?

Comment (cipolleschi) [reply solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListViewController.swift:82]: A router could work. However, we can use something more lightweight: a `Router` requires us to create an abstraction and a source view controller, on which the `Router` can invoke the `present` method. Something like:
```swift
protocol Router {
  func navigate(from viewController: UIViewcontroller)
}

struct ModalSettingsRouter: Router {
  func navigate(from viewController: UIViewController) {
    viewController.present( /* settings view controller code */)
  }
}
```

We can avoid completely to create these additional abstractions and objects if we use a simple closure.

The list view controller can take a closure called `onSettingsTapped` in the init and the `SceneDelegate` can implement the closure to present a proper `ViewController`.

The final solution will be something like this:

```swift 
class ListViewController: UIViewController {
  let onSettingsTapped: (UIViewController) -> Void

  init(onSettingsTapped: @escaping (UIViewController) -> Void) {
    self.onSettingsTapped = onSettingsTapped
  }

  @objc private func showSettings() {
    self.onSettingsTapped(self) // or we can unwrap the navigation controller
  }
}

class SceneDelegate {

  func scene(_ scene: UIScene, willConnectTo _: UISceneSession, options _: UIScene.ConnectionOptions) {
    // other part of the code
    let vc = NavigationController(rootViewController: ListViewController(onSettingsTapped: { presentingVC in
      presentingVC.present(SettingsViewController()) // and all its configurations
    }))
  }
}
```

By doing this, not only the `ListViewController` does not have to know anything about the `SettingsViewController`, but it does not have to know about all the interactions and other pieces of the pattern. Plus, we can decide how to present the SettingsViewController (push or present) without changing the implementation of the ListViewController.

What do you think? Does it make sense? Do you like it? 😊

**Comment by rdgborges** [comment]
Please, reopen if needed.
